strawberryfields/backends/tfbackend/ops.py

```diff
@@ -172,44 +195,31 @@ def squeezed_vacuum_vector(r, theta, D, batched=False, eps=1e-32):
     return output
 
 
-def squeezer_matrix(r, theta, D, batched=False):
-    """creates the single mode squeeze matrix"""
-    r = tf.cast(r, tf.float64)
-    if not batched:
-        r = tf.expand_dims(r, 0)  # introduce artificial batch dimension
-    r = tf.reshape(r, [-1, 1, 1, 1])
-    theta = tf.cast(theta, def_type)
-    theta = tf.reshape(theta, [-1, 1, 1, 1])
+@tf.custom_gradient
+def single_squeezing_matrix(r, phi, D, dtype=def_type.as_numpy_dtype):
+    """creates a single-mode squeezing matrix"""
+    r = r.numpy()
+    phi = phi.numpy()
+    gate = squeezing_tw(r, phi, D, dtype)
```

Comment:
It took me a while to figure out that `D` stands for the `cutoff`. Maybe it would be good to use a more descriptive name?

Reply:
Yes, I agree. I would call it `cutoff`. Although throughout the codebase the name for `cutoff` is not consistent.